app/src/main/kotlin/com/adesso/movee/scene/movie/MovieViewModel.kt

```diff
@@ -49,12 +49,8 @@ class MovieViewModel @Inject constructor(
     }
 
     private fun fetchNowPlayingMovies() {
-        bgScope.launch {
-            val nowPlayingMoviesResult = fetchNowPlayingMoviesUseCase.run(UseCase.None)
-
-            onUIThread {
-                nowPlayingMoviesResult.either(::handleFailure, ::postNowPlayingMovieList)
-            }
+        fetchNowPlayingMoviesUseCase.invoke(viewModelScope, UseCase.None) {
+            it.either(::handleFailure, ::postNowPlayingMovieList)
```

Comment:
postNowPlayingMovieList() uses LiveData's setValue() function which must be called from main thread. Either we need to use postValue(), or switch context when calling postNowPlayingMovieList()